service/controller/v25/resource/encryptionkey/create.go

```diff
@@ -2,33 +2,48 @@ package encryptionkey
 
 import (
 	"context"
+	"time"
 
+	"github.com/giantswarm/backoff"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/aws-operator/service/controller/v25/controllercontext"
 	"github.com/giantswarm/aws-operator/service/controller/v25/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
-	customObject, err := key.ToCustomObject(obj)
+	cr, err := key.ToCustomObject(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
-
-	err = r.encrypter.EnsureCreatedEncryptionKey(ctx, customObject)
+	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	encryptionKey, err := r.encrypter.EncryptionKey(ctx, customObject)
+	err = r.encrypter.EnsureCreatedEncryptionKey(ctx, cr)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	cc, err := controllercontext.FromContext(ctx)
-	if err != nil {
-		return microerror.Mask(err)
+	var encryptionKey string
+	{
+		o := func() error {
```

Comment:
I think it would be good to drop the comment that the key isn't immediately available after the creation in case someone will try to optimize the resource in the future.

Reply:
I added the comment from the PR description.